addons/binding/org.openhab.binding.wso2iots/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="wso2iots"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="buildingMonitor">
+	    <supported-bridge-type-refs>
+            <bridge-type-ref id="wso2iotsBridge" />
+        </supported-bridge-type-refs>
+        
+		<label>Building Monitor</label>
+		<description>Provides temperature, motion, humidity, light level collected from building monitoring devices</description>
+
+		<channels>
+			<channel id="temperature" typeId="temperature" />
+			<channel id="humidity" typeId="humidity" />
+			<channel id="motion" typeId="motion" />
+			<channel id="light" typeId="light" />
+		</channels>
+
+		<config-description>
+			<parameter name="deviceId" type="text" required="true">
+				<label>Device ID</label>
+				<description>ID of the building monitor device</description>
+			</parameter>
+
+			<parameter name="apikey" type="text" required="true">
+				<context>password</context>
+				<label>API Key</label>
+				<description>Data-platform token to access the IOT server</description>
+			</parameter>
+
+			<parameter name="refresh" type="integer" min="2" required="true">
+				<label>Refresh interval</label>
+				<description>Specifies the refresh interval in minutes.</description>
+				<default>2</default>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="temperature">
+		<item-type>Number</item-type>
+		<label>Temperature</label>
+		<description>Temperature in Celsius degrees</description>
+		<category>Temperature</category>
+		<state readOnly="true" pattern="%.1f °C" />
```

Comment:
Consider implementing support for Units of Measurment, see a lot of recent code.